### DISCUSSION TITLE
Welcome to flying-brick Discussions!

### DISCUSSION THREAD

**Discussion by tml1024** [tml1024/flying-brick#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Prepar3DGuy** [comment]
Hi. I am following your project after announcement on fsdeveloper.com. First of all thank you for sharing you project with community. By exploring your project and making some changes to test ideas I'm motivating to learn modern C++ and making fresh look at the old problems.

As a hobby on my leisure I'm trying to make airplane for flight simulator with custom flight model, studying flight simulator addon development and helping people on forum when I can.

As I remember from old days of FSX, and P3D (ESP) the implementation of custom flight model in dll-module was possible using FREEZE_ATTITUDE, FREEZE_ALTITUDE, FREEZE_LATITUDE_LONGITUDE and setting writable simulation variables, but have some limitations on performance. As I have found out the primary function for that approach was to implement custom movement of AI objects and changing user airplanes position for missions. Also from the first releases of MSFS SDK I have checked my old code for compatibility. 
Couple years ago P3D SDK has provided opportunity to implement custom model for simobject movement like I have always dreamed. The killer feature was to replace default sim model with custom one give you opportunity to simulate everything that can be move in space, from bee and ant to dolphin swimming under water. And yes, they have default quadrotor implemented that way. Of course that product have different market and purposes, so let it be and evolve. As I understand the MSFS and its predecessors have idea to simplify the process of making new airplanes by parametric model in textual configs. Thanks that awful air-files now obsolete. For now (a year after MSFS release) I see that developers of MSFS are prioritized to airplane flight model only for now and also they will not follow the P3D way.

When setting FREEZE the default simulation doesn't turning off, it's only stop translating internally calculated attitude, altitude, latitude and longitude to user airplane. MSFS continue to simulate wind, gauges, process user inputs, binded animations over time. I have tested on your project that if setting changed altitude and velocity on vertical axis and only requesting acceleration on the same axis then it's in opposite direction and calculated inside the sim. Of course we can also calculate and set SimVars for acceleration and rotation velocity and acceleration. What you think about if this variables will be rewrited by MSFS default model faster then every PERIOD_SIM_FRAME? Your RequestDataOnSimObject for SIMCONNECT_PERIOD_SIM_FRAME are received before or after actual calculation of default MSFS flight model?

Seems that your headache for actual landing of yellow-brick will easy with contact point simvars in SimConnect, that was [announcement](https://www.flightsimulator.com/june-3rd-2021-development-update/) this week . But I have not found it yet in SDK.

**Reply by tml1024** [reply]
I do notice that even if FREEZE_ATTITUDE, FREEZE_ALTITUDE, FREEZE_LATITUDE_LONGITUDE are on, when the callback is called new values for the corresponding velocities have been calculated (pointlessly) by the simulator. But that is just some wasted CPU, it doesn't seem to affect how the aircraft behaves.

**Reply by tml1024** [reply]
Thanks for the hint about the contact point simvars, will have to look for those in next SDK releases. Those will be useful for many kinds of "non-standard" aircraft.

**Comment by bendanville** [comment]
Awesome project! I am interested in something that's a bit of a tangent, does the MSFS SDK support overriding lats, lons etc as I have an existing external flight model I would like to connect to MSFS for visuals
